homeassistant/components/opentherm_gw/const.py

```diff
@@ -57,167 +62,275 @@
-    ],
-    gw_vars.DATA_SLAVE_CH2_PRESENT: [
-        None,
-        "Boiler Central Heating 2 Present {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_SLAVE_SERVICE_REQ: [
-        BinarySensorDeviceClass.PROBLEM,
-        "Boiler Service Required {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_SLAVE_REMOTE_RESET: [
-        None,
-        "Boiler Remote Reset Support {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_SLAVE_LOW_WATER_PRESS: [
-        BinarySensorDeviceClass.PROBLEM,
-        "Boiler Low Water Pressure {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_SLAVE_GAS_FAULT: [
-        BinarySensorDeviceClass.PROBLEM,
-        "Boiler Gas Fault {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_SLAVE_AIR_PRESS_FAULT: [
-        BinarySensorDeviceClass.PROBLEM,
-        "Boiler Air Pressure Fault {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_SLAVE_WATER_OVERTEMP: [
-        BinarySensorDeviceClass.PROBLEM,
-        "Boiler Water Overtemperature {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_REMOTE_TRANSFER_DHW: [
-        None,
-        "Remote Hot Water Setpoint Transfer Support {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_REMOTE_TRANSFER_MAX_CH: [
-        None,
-        "Remote Maximum Central Heating Setpoint Write Support {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_REMOTE_RW_DHW: [
-        None,
-        "Remote Hot Water Setpoint Write Support {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_REMOTE_RW_MAX_CH: [
-        None,
-        "Remote Central Heating Setpoint Write Support {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_ROVRD_MAN_PRIO: [
-        None,
-        "Remote Override Manual Change Priority {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.DATA_ROVRD_AUTO_PRIO: [
-        None,
-        "Remote Override Program Change Priority {}",
-        [gw_vars.BOILER, gw_vars.THERMOSTAT],
-    ],
-    gw_vars.OTGW_GPIO_A_STATE: [None, "Gateway GPIO A {}", [gw_vars.OTGW]],
-    gw_vars.OTGW_GPIO_B_STATE: [None, "Gateway GPIO B {}", [gw_vars.OTGW]],
-    gw_vars.OTGW_IGNORE_TRANSITIONS: [
-        None,
-        "Gateway Ignore Transitions {}",
+
+@dataclass(frozen=True)
+class OpenThermEntityDescriptionMixin:
+    """Mixin for common opentherm_gw entity properties."""
+
+    friendly_name_format: str
+
+
+@dataclass(frozen=True, kw_only=True)
+class OpenThermBinarySensorEntityDescription(
+    BinarySensorEntityDescription, OpenThermEntityDescriptionMixin
+):
+    """Describes opentherm_gw binary sensor entity."""
+
+
+BINARY_SENSOR_INFO: tuple[
+    tuple[list[str], OpenThermBinarySensorEntityDescription], ...
+] = (
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
```

Comment:
These should not live in `const.py` as these are binary_sensor only things, please move them to `binary_sensor.py`

Reply:
That makes sense, however I plan to use the mixin for sensor entities as well. Should I move it to `binary_sensor.py` now and back here later or can I leave it here now to reduce the size of the follow-up pr?

Reply:
I mean ideally we use translation keys

Reply:
We usually find this in entity.py together with a shared entity

Reply:
Thank you for the feedback, I will separate it into `entity.py`. Translation keys will be in a followup PR.